pydantic/_internal/_decorators.py

```diff
@@ -191,12 +193,32 @@ def __init__(
         self.json_return_type = json_return_type
 
 
+class ModelValidatorDecoratorInfo(Representation):
+    """
+    A container for data from `@model_validator` so that we can access it
+    while building the pydantic-core schema.
+    """
+
+    __slots__ = ('mode',)
+
+    mode: Literal['wrap', 'before', 'after']
+    json_return_type: JsonReturnTypes | None
```

Comment:
can be removed.

As mentioned on #5379, I think we should convert these to dataclasses.

Reply:
I'm confused: you linked to this pull request

Reply:
Nvm, makes sense now. See https://github.com/pydantic/pydantic/pull/5416